src/pages/Login.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import Typography from '@mui/material/Typography';
+import { Stack, TextField, Button } from '@mui/material';
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+
+export const Login = () => {
+    const apiUrl = import.meta.env.VITE_API_URL;
+    const initialValues = { email: '', password: '' };
+    const [formValues, setFormValues] = useState(initialValues);
+    const [error, setError] = useState<string | null>(null);
+    const [success, setSuccess] = useState<string | null>(null);
+    const navigate = useNavigate();
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setFormValues((prev) => ({ ...prev, [name]: value }));
+    };
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+        if (!emailRegex.test(formValues.email)) {
+            setError('Please provide a valid email.');
+            return;
+        }
+
+        try {
+            await axios.post(`${apiUrl}/users/login`, formValues);
+            setSuccess('Login successful!');
+            setTimeout(() => {
+                setSuccess('');
+                navigate('/');
+            }, 1000);
+            setError(null);
+        } catch (err) {
+            console.log('catched error: ', err);
+            let messageError;
+            if (
+                axios.isAxiosError(err) &&
+                err.response &&
+                err.response.data &&
+                err.response.data.message
+            ) {
+                messageError = err.response.data.message;
+            } else {
+                messageError = 'An error occurred while logging in.';
+            }
+            setError(messageError);
+            setSuccess(null);
+        }
+    };
+
+    return (
+        <Stack
+            component='form'
+            onSubmit={handleSubmit}
+            spacing={4}
+            paddingX={3}
+            paddingY={5}
+            direction='row'
+            justifyContent='center'
+            sx={{
+                marginTop: '100px',
+                marginX: 'auto',
+                width: { xs: '100%', sm: '450px' },
+                borderRadius: 4,
+                boxShadow: 3,
+            }}
+        >
+            <Stack spacing={3}>
+                <Typography variant='h4' component='h1'>
+                    Login form
+                </Typography>
+                {error && (
+                    <Typography color='error'>
+                        {Array.isArray(error) ? error.join(', ') : error}
+                    </Typography>
+                )}
+                {success && (
+                    <Typography color='success.main'>{success}</Typography>
+                )}
+                <TextField
```

Comment:
<TextField
          error
          id="outlined-error-helper-text"
          label="Error"
          defaultValue="Hello World"
          helperText="Incorrect entry."
        />
        
        https://mui.com/material-ui/react-text-field/#validation
        
        You can use error and helperText to show validation information in a field.

Reply:
Thank you, I used error and helperText